### DISCUSSION TITLE
Setup and Login to PhotoPrism for first time

### DISCUSSION THREAD

**Discussion by leocordis27** [photoprism/photoprism#2666] [General]
Hi all,

I have been followin wundertech website on how to setup PhotoPrism for the first time.
https://www.wundertech.net/how-to-setup-photoprism-on-a-synology-nas/#1_Setup_Instructions

![image](https://user-images.githubusercontent.com/111774381/187595247-1d97a11d-1105-4b9b-8510-cb3b5bbaae4d.png)

The problem is my docker environment page is different (maybe newer version?) and it shows four environment that I need to fill in. What should I fill? And do I need to add the "PHOTOPRISM_ADMIN_PASSWORD" as well?

![image](https://user-images.githubusercontent.com/111774381/187595365-57bfcc8c-5976-4878-b2cf-87887493ffaf.png)

If I don't change the environment, Photoprism still works but I don't know the admin and password to login.

![image](https://user-images.githubusercontent.com/111774381/187595565-fd962433-1dbf-4fb7-98a5-30af9c27d2d2.png)

What should I do regarding the environment?

Thanks!

**Comment by graciousgrey** [comment]
Unfortunately we don't have a Synology device for testing. 

Maybe another Synology user can help.

In any case the admin password is needed, otherwise you can't sign in.

Site author and description are optional. 

For the database related variables it depends on whether you want to use sqlite or mariadb. Use SQLite only for small libraries and testing. SQLite locks the index on updates, so other operations have to wait. In the worst case, this can lead to timeout errors. MariaDB is optimized for high concurrency.

Example configurations can be found in our docker-compose.yml examples:

 **Mariadb**: https://dl.photoprism.app/docker/docker-compose.yml

      PHOTOPRISM_DATABASE_DRIVER: "mysql"            # use MariaDB 10.5+ or MySQL 8+ instead of SQLite for improved performance
      PHOTOPRISM_DATABASE_SERVER: "mariadb:3306"     # MariaDB or MySQL database server (hostname:port)
      PHOTOPRISM_DATABASE_NAME: "photoprism"         # MariaDB or MySQL database schema name
      PHOTOPRISM_DATABASE_USER: "photoprism"         # MariaDB or MySQL database user name
      PHOTOPRISM_DATABASE_PASSWORD: "insecure"       # MariaDB or MySQL database user password

**Sqlite**: https://dl.photoprism.app/docker/sqlite/docker-compose.yml

      PHOTOPRISM_DATABASE_DRIVER: "sqlite"           # SQLite is an embedded database that doesn't require a server

**Reply by leocordis27** [reply]
I don't think I have ever never setup mariaDB via docker before. Does this mean I need to install mariaDB via docker first?

**Reply by graciousgrey** [reply]
If photoprism is installed using the docker-compose method (https://docs.photoprism.app/getting-started/docker-compose/) there is no need to manually setup mariadb, it will work out of the box.

I am not sure how it behaves on Synology devices.
